uniprot-datastore/src/main/java/org/uniprot/store/datastore/uniparc/UniParcStoreStep.java

```diff
@@ -0,0 +1,99 @@
+import org.uniprot.store.job.common.listener.LogRateListener;
+import org.uniprot.store.job.common.listener.WriteRetrierLogStepListener;
+import org.uniprot.store.job.common.writer.ItemRetryWriter;
+
+/**
+ * @author lgonzales
+ * @since 2020-03-03
+ */
+public class UniParcStoreStep {
+    private final StepBuilderFactory stepBuilderFactory;
+    private final UniParcStoreProperties uniParcStoreProperties;
+
+    @Autowired
+    public UniParcStoreStep(
+            StepBuilderFactory stepBuilderFactory, UniParcStoreProperties uniParcStoreProperties) {
+        this.stepBuilderFactory = stepBuilderFactory;
+        this.uniParcStoreProperties = uniParcStoreProperties;
+    }
+
+    @Bean(name = "uniParcStoreMainStep")
+    public Step uniParcStoreMainStep(
+            WriteRetrierLogStepListener writeRetrierLogStepListener,
+            @Qualifier("uniParcLogRateListener")
+                    LogRateListener<UniParcEntry> uniParcLogRateListener,
+            ItemReader<Entry> entryItemReader,
+            ItemProcessor<Entry, UniParcEntry> uniParcEntryProcessor,
+            ItemWriter<UniParcEntry> uniParcEntryItemWriter,
+            ExecutionContextPromotionListener promotionListener)
+            throws Exception {
+
+        return this.stepBuilderFactory
+                .get(UNIPARC_STORE_STEP)
+                .listener(promotionListener)
+                .<Entry, UniParcEntry>chunk(uniParcStoreProperties.getChunkSize())
+                .reader(entryItemReader)
+                .processor(uniParcEntryProcessor)
+                .writer(uniParcEntryItemWriter)
+                .listener(writeRetrierLogStepListener)
+                .listener(uniParcLogRateListener)
+                .listener(unwrapProxy(uniParcEntryItemWriter))
+                .build();
+    }
+
+    // ---------------------- Readers ----------------------
+    @Bean
+    public ItemReader<Entry> uniParcEntryItemReader() {
+        return new UniParcXmlEntryReader(uniParcStoreProperties.getXmlFilePath());
+    }
+
+    // ---------------------- Processors ----------------------
+    @Bean
+    public ItemProcessor<Entry, UniParcEntry> uniParcEntryProcessor() {
+        return new UniParcEntryProcessor();
+    }
+
+    // ---------------------- Writers ----------------------
+    @Bean
+    public ItemRetryWriter<UniParcEntry, UniParcEntry> uniParcEntryItemWriter(
+            UniProtStoreClient<UniParcEntry> uniParcStoreClient,
+            RetryPolicy<Object> writeRetryPolicy) {
+        return new UniParcEntryRetryWriter(
+                entries -> entries.forEach(uniParcStoreClient::saveEntry), writeRetryPolicy);
+    }
+
+    // ---------------------- Listeners ----------------------
+    @Bean(name = "uniParcLogRateListener")
+    public LogRateListener<UniParcEntry> uniParcLogRateListener() {
+        return new LogRateListener<>(uniParcStoreProperties.getLogRateInterval());
+    }
+
+    private Object unwrapProxy(Object bean) throws Exception {
```

Comment:
I am not able to understand the purpose of this method. From the usage, it seems, it is some kind of listener. But it is not obvious.